src/libpriv/rpmostree-refts.cxx

```diff
@@ -163,18 +164,22 @@ RpmTs::package_meta (const rust::Str name) const
   std::unique_ptr<PackageMeta> retval;
   while ((h = rpmdbNextIterator (mi)) != NULL)
     {
-      // TODO: Somehow we get two `libgcc-8.5.0-10.el8.x86_64` in current RHCOS, I don't
-      // understand that.
-      if (retval != nullptr)
+      // 64 characters ought to be enough for any arch
+      if (strncmp (headerGetString (h, RPMTAG_ARCH), arch_c.c_str (), 64) == 0)
```

Comment:
These are both NUL terminated strings, right? Why can't we just use `strcmp()`? (Or if you want, `g_str_equal`)

Reply:
I think at a previous job it was a linter error to use plain `strcmp` and I got used to putting an upper bound on everything. Happy to change it if you prefer.

Reply:
Hmm. That sounds like a linter rule that could easily cause as many problems as it solves... 

First it would only partially cap undefined behavior if somehow you passed it a not-NUL terminated string under the max, and if you have two NUL terminated strings where usually it's short (say, a username or password) but capping at an arbitrary max below the length could cause two strings that only differ past the maximum to compare equal...which could be really annoying to figure out.

The codebase here has a ton of `strcmp()` and equivalent so yes, please do use it.

Reply:
I've changed it